app/ide-desktop/client/src/server.ts

```diff
@@ -227,18 +227,26 @@ export class Server {
           }
 
           https.get(downloadUrl, async (actualResponse) => {
-            try {
-              const projectsDirectory = projectManagement.getProjectsDirectory()
-              const parentDirectory = path.join(projectsDirectory, `cloud-${projectId}`)
-              const targetDirectory = path.join(parentDirectory, 'project_root')
+            const projectsDirectory = projectManagement.getProjectsDirectory()
+            const parentDirectory = path.join(projectsDirectory, `cloud-${projectId}`)
+            const targetDirectory = path.join(parentDirectory, 'project_root')
 
+            try {
               await fs.mkdir(targetDirectory, { recursive: true })
               await projectManagement.unpackBundle(actualResponse, targetDirectory)
               response
                 .writeHead(HTTP_STATUS_OK, COOP_COEP_CORP_HEADERS)
                 .end(JSON.stringify({ targetDirectory, parentDirectory }))
             } catch (e) {
-              console.error(e)
+              logger.error(e)
+              await fs
+                .access(parentDirectory)
+                .then(() => {
+                  fs.rmdir(parentDirectory, { maxRetries: 3, recursive: true })
+                })
+                .catch((e) => {
+                  logger.error(`Failed to cleanup directory ${parentDirectory}.`, e)
```

Comment:
not sure if we already do but ideally we would display this error to the user via something like a toast, @MrFlashAccount WDYT?

Reply:
We already show the toast on errors. This case is when things went completely wrong, and we cannot clean up the directory for some reason. This logging will be useful for debugging